Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -0,0 +1,1230 @@
+                        closestCollider = ballColliders[i];
+                    }
+                }
+            }
+
+            distance = closestDistance;
+            return closestCollider;
+        }
+        private Ray GetHandleGrabbedRay()
+        {
+            Ray pointerRay = new Ray();
+            if (currentInputSource.Pointers.Length > 0)
+            {
+               currentInputSource.Pointers[0].TryGetPointingRay(out pointerRay);
+            }
+
+            return pointerRay;
+        }
+        private void Flatten()
+        {
+            if (flattenAxis == FlattenModeType.FlattenX)
+            {
+                flattenedHandles = new int[] { 0, 4, 2, 6 };
+            }
+            else if (flattenAxis == FlattenModeType.FlattenY)
+            {
+                flattenedHandles = new int[] { 1, 3, 5, 7 };
+            }
+            else if (flattenAxis == FlattenModeType.FlattenZ)
+            {
+                flattenedHandles = new int[] { 9, 10, 8, 11 };
+            }
+
+            if (flattenedHandles != null)
+            {
+                for (int i = 0; i < flattenedHandles.Length; ++i)
+                {
+                    linkRenderers[flattenedHandles[i]].enabled = false;
+                }
+            }
+        }
+        private void SetHiddenHandles()
+        {
+            if (flattenedHandles != null)
+            {
+                for (int i = 0; i < flattenedHandles.Length; ++i)
+                {
+                    ballRenderers[flattenedHandles[i]].enabled = false;
+                }
+            }
+        }
+        private void GetCornerPositionsFromBounds(Bounds bounds, ref Vector3[] positions)
+        {
+            Vector3 center = bounds.center;
+            Vector3 extents = bounds.extents;
+            float leftEdge = center.x - extents.x;
+            float rightEdge = center.x + extents.x;
+            float bottomEdge = center.y - extents.y;
+            float topEdge = center.y + extents.y;
+            float frontEdge = center.z - extents.z;
+            float backEdge = center.z + extents.z;
+
+            if (positions == null || positions.Length != cornerCount)
+            {
+                positions = new Vector3[cornerCount];
+            }
+
+            positions[LBF] = new Vector3(leftEdge, bottomEdge, frontEdge);
+            positions[LBB] = new Vector3(leftEdge, bottomEdge, backEdge);
+            positions[LTF] = new Vector3(leftEdge, topEdge, frontEdge);
+            positions[LTB] = new Vector3(leftEdge, topEdge, backEdge);
+            positions[RBF] = new Vector3(rightEdge, bottomEdge, frontEdge);
+            positions[RBB] = new Vector3(rightEdge, bottomEdge, backEdge);
+            positions[RTF] = new Vector3(rightEdge, topEdge, frontEdge);
+            positions[RTB] = new Vector3(rightEdge, topEdge, backEdge);
+        }
+        private static Vector3 PointToRay(Vector3 origin, Vector3 end, Vector3 closestPoint)
```

Comment:
I'm wondering how much of these methods could go into a math utility class

Reply:
they may be able to be added to a util class if there is one- but i wanted to celebrate the idea of a One-File BB solution for a little while :)

Reply:
since these are private methods, we can move them to a utils class as needed.

Reply:
actually i removed two of the three functions. since there in only the pointtoray snap function- ill leave it in rather than create a new file for just one function.